common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -1527,65 +1527,55 @@ bool ap_wlan_hal_dwpal::read_acs_report()
 
 bool ap_wlan_hal_dwpal::read_supported_channels()
 {
-    char *reply = nullptr;
-    if (!dwpal_send_cmd("GET_HW_FEATURES", &reply)) {
-        LOG(ERROR) << __func__ << " failed";
-        return false;
-    }
+    LOG(TRACE) << __func__ << " for interface: " << get_radio_info().iface_name;
 
-    size_t replyLen = strnlen(reply, HOSTAPD_TO_DWPAL_MSG_LENGTH);
-    for (size_t i = 0; i < replyLen; i++) {
-        if (reply[i] == ')') {
-            reply[i] = ' ';
-        }
+    // Clear the supported channels vector
+    m_radio_info.supported_channels.clear();
+    // Resize the supported channels vector
+    m_radio_info.supported_channels.resize(MAX_SUPPORTED_20M_CHANNELS);
+    bwl::nl80211_client::radio_info radio_info;
```

Comment:
i suggest moving all "m_nl80211_client->get_radio_info" related code to a private function
for example `get_supported_channels_from_nl80211`

Reply:
so we could easily switch back to Hostapd implementation in the future, but since we are not keeping it then it doesn't matter.

Reply:
Currently all read_supported_channels (dwpal & nl80211) will use the
`m_nl80211_client->get_radio_info` instead of the old `GET_HW_FEATURES` & the old NL operation